src/main/java/org/folio/circulation/resources/CheckInProcessAdapter.java

```diff
@@ -93,7 +95,13 @@
   CompletableFuture<Result<Item>> updateItem(CheckInProcessRecords records) {
     return updateItem.onCheckIn(records.getItem(), records.getRequestQueue(),
       records.getCheckInServicePointId(), records.getLoggedInUserId(),
-      records.getCheckInRequest().getCheckInDate());
+      determineCheckInDate(records));
+  }
+
+  private DateTime determineCheckInDate(CheckInProcessRecords records) {
+    DateTime checkInDate = records.getCheckInRequest().getCheckInDate();
+    DateTime now = ClockManager.getClockManager().getDateTime();
+    return checkInDate.isBefore(now) ? now : checkInDate;
```

Comment:
If I'm following the story correctly, the intention is for the `last check in date time` to be when the last check in was processed on the system. Is this a sensible understanding?

If so, why wouldn't this always be now?

Reply:
Yes, it's always now now.

Reply:
Then what is the purpose of the check for before and a decision between two different dates?

Reply:
This check is no more in the latest version of the code.

Reply:
Should this value be consistent with the system return date for a loan?

I think it probably should be, meaning this value should be used as a parameter for the loan check in method.

Reply:
Do you mean it should come from FE - current date time?

Reply:
As I understand it, the value does not come from the client at the moment. I don't think that should be changed to resolve this bug.

I mean that the `last check in date time` and the `system return date time` should likely be the same. And so the value chosen should be used for both.

I think that means that the `checkIn` method should have an extra parameter for the `system return date time` and that should be the same value as used for `last check in date time`.

Does that make sense?

Reply:
Sure, thank you for the explanation.

Reply:
Yes, the tests now use exact comparisons.